### ISSUE TITLE
File streams: "BufferedWriter" has incompatible type "BinaryIO"; expected "RawIOBase"

### ISSUE THREAD

**Issue by h6197627** [python/mypy#10271]
**Bug Report**
This code
```
import io

bf = open('binary.file', 'wb', buffering=0)
print(isinstance(bf, io.RawIOBase)) # > True
bw = io.BufferedWriter(bf)
```
throws:

> test.py:5: error: Argument 1 to "BufferedWriter" has incompatible type "BinaryIO"; expected "RawIOBase"

**Your Environment**

- Mypy version used: 0.812
- Mypy command-line flags:
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used: 3.6.9
- Operating system and version: Ubuntu 18.04

**Comment by h6197627** [comment]
Just checked, I have
```
# Unbuffered binary mode: returns a FileIO
@overload
def open(
    file: _OpenFile,
    mode: OpenBinaryMode,
    buffering: Literal[0],
    encoding: None = ...,
    errors: None = ...,
    newline: None = ...,
    closefd: bool = ...,
    opener: Optional[_Opener] = ...,
) -> FileIO: ...
```
in /usr/local/lib/python3.6/dist-packages/mypy/typeshed/stdlib/3/builtins.pyi

**Comment by srittau** [comment]
I can confirm. With mypy 0.812 it doesn't pick up the correct overload, although it exists in the bundled typeshed and it looks like it should pick it.

**Comment by JelleZijlstra** [comment]
It's actually because mypy still has a custom plugin for `open` that produces less precise types. #9275 removes it, but hasn't been merged yet.

**Comment by h6197627** [comment]
Hi,
any progress on this issue? :)

**Comment by h6197627** [comment]
Fixed with #9275

**Comment by lifepillar** [comment]
I can still reproduce it with mypy 0.931:

```sh
$ mypy test.py
foo.py:5:24: error: Argument 1 to "BufferedWriter" has incompatible type "BinaryIO"; expected "RawIOBase"
Found 1 error in 1 file (checked 1 source file)
$ mypy --version
mypy 0.931
```
